Assets/MRTK/Providers/WindowsMixedReality/Shared/WindowsMixedRealityArticulatedHandDefinition.cs

```diff
@@ -123,14 +126,23 @@ public void UpdateHandMesh(SpatialInteractionSourceState sourceState)
 
                 if (handMeshObserver != null && handPose != null)
                 {
-                    if (handMeshTriangleIndices == null)
+                    if ((handMeshTriangleIndices == null) ||
+                        (handMeshTriangleIndices.Length != handMeshObserver.TriangleIndexCount))
                     {
                         handMeshTriangleIndices = new ushort[handMeshObserver.TriangleIndexCount];
                         handMeshTriangleIndicesUnity = new int[handMeshObserver.TriangleIndexCount];
+                    }
+
+                    if (!HandMeshModelId.HasValue || (HandMeshModelId.Value != handMeshObserver.ModelId))
```

Comment:
Can this be simplified to
```suggestion
                    if (HandMeshModelId != handMeshObserver.ModelId)
```

Reply:
Not if we use nullable ints.

Reply:
but `null` should not equal any int, right? I thought C# was smart about checking nullable types against non-nullable ones :(

Reply:
i may be thinking "old" c# 😄